rllib/algorithms/ppo/tests/test_ppo_with_rl_module.py

```diff
@@ -1,18 +1,93 @@
-import numpy as np
 import unittest
 
+import numpy as np
+import tree
+import gym
+import torch
+
 import ray
 import ray.rllib.algorithms.ppo as ppo
-
 from ray.rllib.algorithms.callbacks import DefaultCallbacks
+from ray.rllib.algorithms.ppo.torch.ppo_torch_rl_module import (
+    PPOModuleConfig,
+    PPOTorchRLModule,
+    get_ppo_loss,
+)
+from ray.rllib.core.rl_module.encoder import (
+    IdentityConfig,
+    FCConfig,
+    LSTMConfig,
+)
 from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+from ray.rllib.policy.sample_batch import SampleBatch
 from ray.rllib.utils.metrics.learner_info import LEARNER_INFO, LEARNER_STATS_KEY
+from ray.rllib.utils.numpy import convert_to_numpy
 from ray.rllib.utils.test_utils import (
     check,
     check_compute_single_action,
     check_train_results,
     framework_iterator,
 )
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+
+
+# Get a model config that we expect from the catalog
+def get_expected_model_config(env, lstm, shared_encoder):
```

Comment:
can we make this part of the PPORLModule's `from_model_config` api? putting this utility function in the unittest file will cause information leakage between the implementation details of PPORLModule and the test. If PPORLModule changes due to catalog stuff getting developed then this test will also have to migrate and it will break. But if you put this logic inside `from_model_config` then that API has to get fixed as part of the catalog stuff and this test should not change really.